### PR TITLE
ARIA-299 Resuming canceled execution with frozen task fails

### PR DESCRIPTION
**Pull request by mxmrlv**

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

File: aria/orchestrator/workflows/core/events_handler.py [hunk 1]
```diff
@@ -114,6 +114,10 @@ def _workflow_cancelled(workflow_context, *args, **kwargs):
         elif execution.status in (execution.SUCCEEDED, execution.FAILED):
             _log_tried_to_cancel_execution_but_it_already_ended(workflow_context, execution.status)
         else:
+            # Any non ended task would be put back to pending state
```
Comment (mxmrlv) [inline aria/orchestrator/workflows/core/events_handler.py]: move to start

File: tests/orchestrator/test_workflow_runner.py [hunk 2]
```diff
@@ -441,8 +509,42 @@ def mock_resuming_task(ctx):
...
-        if not events['is_resumed'].isSet():
+        custom_events['is_active'].set()
+        if not custom_events['is_resumed'].isSet():
             # if resume was called, increase by one. o/w fail the execution - second task should
             # fail as long it was not a part of resuming the workflow
             raise BaseException("wasn't resumed yet")
+
+
+@workflow
+def mock_sequential_workflow(ctx, graph):
```
Comment (mxmrlv) [inline tests/orchestrator/test_workflow_runner.py]: rename

File: tests/orchestrator/test_workflow_runner.py [hunk 3]
```diff
@@ -441,8 +509,42 @@ def mock_resuming_task(ctx):
...
+        # unfreeze the thread only when all of the invocations are done
+        while ctx.node.attributes['invocations'] < ctx.task.max_attempts - 1:
+            pass
+
+    elif ctx.node.attributes['invocations'] == ctx.task.max_attempts - 1:
+        # pass only just before the end.
+        return
+    else:
+        # fail o.w.
+        raise BaseException("stop this task")
```
Comment (mxmrlv) [inline tests/orchestrator/test_workflow_runner.py]: new exception

File: tests/orchestrator/test_workflow_runner.py [hunk 4]
```diff
@@ -441,8 +509,42 @@ def mock_resuming_task(ctx):
...
+    The task should run atmost ctx.task.max_attempts - 1 times, and only then pass.
+    overall, the number of invocations should be ctx.task.max_attempts - 1
+    """
+    ctx.node.attributes['invocations'] += 1
+
+    if ctx.node.attributes['invocations'] == 2:
+        custom_events['is_active'].set()
+        # unfreeze the thread only when all of the invocations are done
+        while ctx.node.attributes['invocations'] < ctx.task.max_attempts - 1:
+            pass
```
Comment (mxmrlv) [inline tests/orchestrator/test_workflow_runner.py]: sleep(5)

File: tests/orchestrator/test_workflow_runner.py [hunk 5]
```diff
@@ -441,8 +509,42 @@ def mock_resuming_task(ctx):
...
+        api.task.OperationTask(node,
+                               interface_name='aria.interfaces.lifecycle',
+                               operation_name='create',
+                               retry_interval=1,
+                               max_attempts=10),
+    )
+
+
+@operation
+def mock_failed_first_task(ctx):
```
Comment (mxmrlv) [inline tests/orchestrator/test_workflow_runner.py]: add test to check if started is resumable

**Pull request merged**